tests: arch: arm: move test arm_hardfault_validation to new ztest API

shaoanxli opened the pull request:
Migrate the testsuite tests/arch/arm/arm_hardfault_validation to the new ztest API.

Signed-off-by: Shaoan Li <shaoanx.li@intel.com>

stephanosio reviewed (changes requested):
The changes look ok to me.

> Signed-off-by: shaoanxli shaoanx.li@intel.com

Please use your full name in the sign-off.

stephanosio reviewed (changes requested):
Please fix the following failures:

```
57baab5d7a114ad31ffeb7372e66ace7ec3b806e: author email (shaoanxli <shaoanx.li@intel.com>) needs to match one of the signed-off-by entries.
57baab5d7a114ad31ffeb7372e66ace7ec3b806e: author email (shaoanxli <shaoanx.li@intel.com>) does not follow the syntax: First Last <email>.
```

```
3: UC4 Line exceeds max length (83>75): "Migrate the testsuite tests/arch/arm/arm_hardfault_validation to the new ztest API."
```

shaoanxli committed:
tests: arch: arm: move test arm_hardfault_validation to new ztest API

Migrate the testsuite tests/arch/arm/arm_hardfault_validation to the
new ztest API.

Signed-off-by: Shaoan Li <shaoanx.li@intel.com>

shaoanxli commented:
> Please fix the following failures:
> 
> ```
> 57baab5d7a114ad31ffeb7372e66ace7ec3b806e: author email (shaoanxli <shaoanx.li@intel.com>) needs to match one of the signed-off-by entries.
> 57baab5d7a114ad31ffeb7372e66ace7ec3b806e: author email (shaoanxli <shaoanx.li@intel.com>) does not follow the syntax: First Last <email>.
> ```
> 
> ```
> 3: UC4 Line exceeds max length (83>75): "Migrate the testsuite tests/arch/arm/arm_hardfault_validation to the new ztest API."
> ```

Have fixed. Thanks.
